pandas/core/indexes/base.py

```diff
@@ -2211,13 +2211,49 @@ def sortlevel(self, level=None, ascending=True, sort_remaining=None):
         return self.sort_values(return_indexer=True, ascending=ascending)
 
     def shift(self, periods=1, freq=None):
-        """
-        Shift Index containing datetime objects by input number of periods and
-        DateOffset
+        """Shift index by desired number of time frequency increments.
+
+        This method is for shifting the values of datetime-like indexes
+        by a specified time increment a given number of times.
+
+        Parameters
+        ----------
+        periods : int
+            Number of periods (or increments) to shift by,
+            can be positive or negative (default is 1).
+        freq : pandas.DateOffset, pandas.Timedelta or string
+            Frequency increment to shift by (default is None).
+            If None, the index is shifted by its own `freq` attribute.
+            Offset aliases are valid strings, e.g., 'D', 'W', 'M' etc.
 
         Returns
         -------
-        shifted : Index
+        pandas.Index
+            shifted index
+
+        Examples
+        --------
+        >>> month_starts = pd.date_range('1/1/2011', periods=5, freq='MS')
+
```

Comment:
Can you show the output here? (so it is easier to compare the result of the shift with the original)